src/Workspaces/Core/Portable/Shared/Extensions/INamespaceOrTypeSymbolExtensions.cs

```diff
@@ -37,24 +35,34 @@ public static IEnumerable<IPropertySymbol> GetIndexers(this INamespaceOrTypeSymb
                 : symbol.GetMembers(WellKnownMemberNames.Indexer).OfType<IPropertySymbol>().Where(p => p.IsIndexer);
         }
 
-        public static int CompareTo(this INamespaceOrTypeSymbol n1, INamespaceOrTypeSymbol n2)
-        {
-            var names1 = s_namespaceOrTypeToNameMap.GetValue(n1, s_getNamePartsCallBack);
-            var names2 = s_namespaceOrTypeToNameMap.GetValue(n2, s_getNamePartsCallBack);
-
-            return CompareNameParts(names1, names2);
-        }
-
         public static IReadOnlyList<string> GetNameParts(this INamespaceOrTypeSymbol symbol)
-        {
-            return s_namespaceOrTypeToNameMap.GetValue(symbol, s_getNamePartsCallBack);
-        }
+            => s_namespaceOrTypeToNameMap.GetValue(symbol, s_getNamePartsCallBack);
 
-        public static int CompareNameParts(IReadOnlyList<string> names1, IReadOnlyList<string> names2)
+        public static int CompareNameParts(
+            IReadOnlyList<string> names1, IReadOnlyList<string> names2, 
+            bool placeSystemNamespaceFirst)
         {
             for (var i = 0; i < Math.Min(names1.Count, names2.Count); i++)
             {
-                var comp = names1[i].CompareTo(names2[i]);
+                var name1 = names1[i];
+                var name2 = names2[i];
+
+                if (i == 0 && placeSystemNamespaceFirst)
```

Comment:
How does this relate to the code we have that does this for sort usings? We have two implementations?